### PR TITLE
/var/run is deprecated. Support using /run

### PR DESCRIPTION
**Pull request by drieskimpe**
- Allow user to configure listen location for daemon
- Install env.d file for location

### CODE REVIEW

**Comment by jfrazelle** [comment]
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

`git commit --amend -s --no-edit && git push -f`

**Comment by tiborvass** [comment]
Ping @tianon

**Comment by tianon** [comment]
Wouldn't it be better to fix this in Docker itself somehow, maybe by detecting whether `/var/run` exists and using `/run` directly if it doesn't?

**Comment by xiaods** [comment]
@tianon 's suggest is prefer, any update?

File: contrib/init/openrc/docker.confd [hunk 1]
```diff
@@ -11,3 +11,7 @@
 
 # any other random options you want to pass to docker
 DOCKER_OPTS=""
+
+# how to contact the docker daemon
+DOCKER_HOST="unix:///run/docker.unix"
```
Comment (thaJeztah) [inline contrib/init/openrc/docker.confd]: Is there a reason to rename it from `docker.sock` to `docker.unix`?

**Comment by thaJeztah** [comment]
Even if docker would detect the right location, I expect a *huge* number of examples that exist on the internet to break if the location changes. Is there a way to (for example) create a symlink in the old location as well, to keep backward-compatibility?

If this is accepted, various parts in documentation have to be updated as well

**Comment by icecrime** [comment]
@drieskimpe Hey! Thanks for the contribution: can you please address @tianon comments? I'm going to close in the meantime as this cannot be merged as it is, but please feel free to update us and we'll reopen if necessary.

**Pull request closed without merging**